### ISSUE TITLE
[QwikCity] Vercel deployment strategy

### ISSUE THREAD

**Issue by Schniz** [BuilderIO/qwik#906]
### Is your feature request related to a problem?

It's not a problem but a real feature request. The current middleware implementations QwikCity is providing allow to manually craft a working POC. However, it's not very responsive and feels kinda hacky.

### Describe the solution you'd like

* being able to start a QwikCity project with Vercel as the deployment target
* that project can run either on Edge Functions or Serverless Functions (depends on the APIs people want to use). Ideally, being able to choose different runtimes based on the endpoint (like Next.js is allowing)
* on local dev, when using the Edge runtime, we can use [`edge-runtime`](https://github.com/vercel/edge-runtime) to enable developers to have a fast and correct environment that targets their production runtime

### Describe alternatives you've considered

I manually built a POC, more on that in "Additional Context".

### Additional context

tl;dr, I believe that the Vercel adapter should work very close to what SvelteKit's Vercel adapter does, by leveraging the [Build Output API](https://vercel.com/docs/build-output-api/v3) that enables full control on the Vercel deployment.

I have a small POC that reuses the Cloudflare middleware with some changes & hacks. I had this before #874 was released so now I can use the Netlify middleware instead, which does not embed the caching strategy, with minor changes as well.

Some of the hacks I needed to implement:

* uses Cloudflare middleware & replace `caches.open` with a function that generates an empty cache as Vercel does not support `caches` right now. I would've used the Netlify middleware but the commit that fixes the stream (#874) by using Uint8Array instead of string wasn't released when I built that POC

* Replaces `fetch` with a function that allows fetching non-fully-qualified URLs to make `useEndpoint` work (the `useEndpoint` function fetches `/my-page` instead of `https://my-url/my-page`). This information exists in the incoming `Request` so I wonder if we should use it if it exists and always opt for fully-qualified URLs in Qwik requests? This makes ton of sense in terms of SSR & Edge imo. wdyt?

* I used `esbuild.define` in the Vite config to replace all `global` usages with `globalThis`, which are more accurate (it's a standard binding afaik). Maybe this should be changed across Qwik?

Moreover, Vercel Edge Functions are limited in size. It would be best if we could've had something like [SvelteKit Vercel adapter's](https://github.com/sveltejs/kit/blob/master/packages/adapter-vercel/README.md#usage) "split" configuration, which outputs a bundle for each endpoint. I understand that this would be a massive change, but I think it will make better DX and better TTFB (less parsing to do, less routing to do as we will offload it to Vercel's routing with `config.json`, etc)
